packages/nimble-components/src/rich-text-editor/specs/README.md

```diff
@@ -0,0 +1,405 @@
+    a long nested lists that is beyond the current width (set by the client), a horizontal scrollbar will appear within the text area to view the content.
+-   The client will determine the maximum height and width of the component. When the client does not override sizing, the component will have a default height and width.
+-   The `formatting toolbar` in the footer section will occupy space based on the number of formatting buttons used. For the initial scope of this
+    component, four formatting buttons will be included, following standard size and spacing guidelines. The `footer-actions` section will occupy the remaining
+    space in the footer.
+
+_Note_: This initial component design serves as a starting point for implementation, and it may undergo changes once the visual design is completed.
+
+### Anatomy
+
+_Shadow DOM template_
+
+```html
+<template>
+    <section id="editor"></section>
+    <footer>
+        <section id="toolbar">
+            <nimble-toolbar>
+                <nimble-toggle-button></nimble-toggle-button>
+            </nimble-toolbar>
+            <slot name="footer-actions"></slot>
+        </section>
+    </footer>
+</template>
+```
+
+_Slot Names_
+
+-   `footer-actions`:
+    1. It is a container that allows a client to easily place buttons at the right bottom of the component to interact with the editor.
+    2. If no content is slotted in the `footer-actions`, the element will be emptied and shrunk to accommodate buttons from the
+       toolbar.
+    3. If the content is slotted in the footer-actions, it will occupy a maximum of thirty percent of the entire horizontal footer. If
+       there are additional elements beyond this limit, they will be positioned below within the same footer-actions container.
+
+_Note_: The positioning of these slot elements in the mobile view of the component has not yet been confirmed.
+
+_Host Classes_
+
+-   none
+
+_Slotted Content/Slotted Classes_
+
+-   none
+
+_CSS Parts_
+
+-   none
+
+### `nimble-rich-text-viewer`
+
+The `nimble-rich-text-viewer` is used for viewing rich text content when a markdown string is passed to it. It performs the post-processing
+tasks to convert the markdown string to corresponding HTML nodes for each text formatting.
+
+### API
+
+_Props/Attrs_
+
+-   `markdownValue` - is an accessor used to get and set the markdown value.
+    -   `getter` - this will merely return the markdown string that is set to the component.
+    -   `setter` - this will parse the markdown string into a Node using
+        [prosemirror-markdown parser](https://github.com/ProseMirror/prosemirror-markdown/blob/master/src/from_markdown.ts#L199) and convert to an HTML to
+        render it in the component section.
+
+_Methods_
+
+-   none
+
+_Events_
+
+-   none
+
+_CSS Classes and CSS Custom Properties that affect the component_
+
+-   The height of the component will always grow to fit the content.
```

Comment:
Yea as mentioned above I think growing the height should be opt-in. It should have the same sizing behavior as the editor.